File: integration_tests/java/ref/Foo.java
package com.jsontypedef.jtdcodegendemo;




public class Foo {

  @JsonValue
  
  private Bar value;


  
  public Foo() {
  }
  


  public Bar getValue() {
    return value;
  }

  public void setValue(Bar value) {
    this.value = value;
  }

}

File: integration_tests/csharp/ref/Foo.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace Jtd.JtdCodegenDemo
{

    
    public class Foo 
    {



        
        public Bar Value { get; set; }



        private class JsonConverter : Newtonsoft.Json.JsonConverter
        {
            public override bool CanRead => true;
            public override bool CanWrite => true;

            public override bool CanConvert(System.Type objectType)
            {
                return objectType == typeof(Bar);
            }

            public override object ReadJson(JsonReader reader, System.Type objectType, object existingValue, JsonSerializer serializer)
            {
                return new Foo { Value = serializer.Deserialize<Bar>(reader) };
            }

            public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
            {
                serializer.Serialize(writer, ((Foo) value).Value);
            }
        }



    }
}
